module/zfs/dmu_send.c

```diff
@@ -1783,7 +1783,9 @@ send_reader_thread(void *arg)
 		case OBJECT:
 		case OBJECT_RANGE:
 		case REDACT: // Redacted blocks must exist
-			bqueue_enqueue(outq, range, sizeof (*range));
+			/* For OBJECT */
+			spill = piggyback_unmodified_spill(srta, range);
+			bqueue_enqueue(outq, range, sizeof (*range) + spill);
```

Comment:
I'm sorry if I miss something, but here you are adding size of the structure and logical size of a spill block.  Aside of those being unrelated units, I am not sure the last one is needed at all.  Could you please explain?

Reply:
This is what DATA type does.

Reply:
Hmm. OK. I suppose original author was just avoiding unbound queue, not really bothering about actual size for smaller non-data records.